src/signal.rs

```diff
@@ -538,22 +489,20 @@ mod tests {
         // We install a signal handler for the specified signal; otherwise the whole process will
         // be brought down when the signal is received, as part of the default behaviour. Signal
         // handlers are global, so we install this before starting the thread.
-        unsafe {
-            register_signal_handler(0, SignalHandler::Siginfo(handle_signal), true, SA_SIGINFO)
-                .expect("failed to register vcpu signal handler");
-        }
+        register_signal_handler(SIGRTMIN(), handle_signal)
+            .expect("failed to register vcpu signal handler");
 
         let killable = thread::spawn(|| loop {});
 
-        let res = killable.kill(SIGRTMAX());
+        let res = killable.kill(SIGRTMAX() + 1);
```

Comment:
Is `SIGRTMAX()` usable? As in the first unusable is `SIGRTMAX()+1`?

Reply:
Hadn't found any resource that discourages usage of SIGRTMIN, respectively of SIGRTMAX, as signums for registration of signal handlers. The [signal(7)](http://man7.org/linux/man-pages/man7/signal.7.html) doc does not discourage the usage of SIGRTMIN, or SIGRTMAX, but expresses the need to use checks to avoid the case when `SIGRTMIN+n` exceeds `SIGRTMAX`.

As a seprate note, at this specific line, we have to use a signum which exceeds `SIGRTMAX()` because we look for the error thrown from `validate_signal_num` function.

This line could've been in the past as simple as:
 ```suggestion
        let res = killable.kill(16);
```

`SIGRTMAX()` usage, as an argument, is superfluous and does not mean that `killable.kill()` call will explicitly raise the `SIGRTMAX()` signal for the specific thread, but instead it would've raised the signal for the signum `SIGRTMIN() + SIGRTMAX()`, which is not a valid signal number also. While in the new case, with the simplifications from this PR, it is necessary to make the call for `killable.kill()` with an argument which explicitly exceeds `SIGRTMAX()`, so that is why I chose `SIGRTMAX() + 1`.